### PR TITLE
Add tutorial notebook for public released truth-match catalog

### PR DESCRIPTION
**Pull request by yymao**
This PR adds a tutorial notebook for public released truth-match catalog. Similar to #1, one needs to get GCR v0.9.2 and also gcr-catalogs master + LSSTDESC/gcr-catalogs#523 + LSSTDESC/gcr-catalogs#524 for this notebook to run. 

The notebook can also use some polish and maybe a few more science examples. I am running of great ideas at this moment so I am opening this PR for people to review and make suggestions.

### CODE REVIEW

File: notebooks/truth_match_table.ipynb [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+   "source": [
+    "## Access Truth-match Table with GCRCatalogs\n",
+    "\n",
+    "Just like the Object Table, the Truth-match Table is availble as parquet files, \n",
+    "and `GCRCatalogs` provides a high-level user interface for accessing it. \n",
+    "\n",
+    "However, while the Truth-match Table is a single set of parquet files, it appears as two entries in `GCRCatalogs` as we will see below. \n",
+    "\n",
+    "- The catalog entry `desc_dc2_run2.2i_dr6_truth` contains all truth information, including true sources that are not detected in the DR6 Object Table. \n",
+    "- The catalog entry `desc_dc2_run2.2i_dr6_object_with_truth_match` contains all the identical information of the Object Table (`desc_dc2_run2.2i_dr6_object`), plus additional columns that contain information of best-match truth objects. In other words, `desc_dc2_run2.2i_dr6_object_with_truth_match` is like a \"left join\" of `desc_dc2_run2.2i_dr6_object` and `desc_dc2_run2.2i_dr6_truth`, but with spatial sepration and magnitude difference as join keys. See the release note for further details. \n",
```
Comment (fjaviersanchez) [inline notebooks/truth_match_table.ipynb]: sepration -> separation

File: notebooks/truth_match_table.ipynb [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+    "\n",
+    "### Which catalog entry should I use?\n",
+    "\n",
+    "- If you don't need any truth information, just use `desc_dc2_run2.2i_dr6_object`. \n",
+    "- If you only need the best-match truth source information for each object (i.e., each row) in the Object Table, use `desc_dc2_run2.2i_dr6_object_with_truth_match`\n",
+    "- If you need more truth infomation beyond the best matches (i.e., undetected soruces, blended sources), you need to use `desc_dc2_run2.2i_dr6_truth`"
```
Comment (fjaviersanchez) [inline notebooks/truth_match_table.ipynb]: soruces -> sources

File: notebooks/truth_match_table.ipynb [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+   "source": [
+    "set(match_cat.list_all_quantities()).difference(object_cat.list_all_quantities())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "These columns describe the infomation of the matched truth source for each object in the Object Table. \n",
+    "Note that most of these columns have a `_truth` postfix to avoid potential confusion with columns in the Object Table.\n",
```
Comment (fjaviersanchez) [inline notebooks/truth_match_table.ipynb]: postfix -> suffix?

File: notebooks/truth_match_table.ipynb [hunk 4]
```diff
@@ -0,0 +1,436 @@
...
+    "In fact, these columns are the same set of columns in `truth_cat` (but there's no `_truth` postfix in `truth_cat`):"
```
Comment (fjaviersanchez) [inline notebooks/truth_match_table.ipynb:167]: postfix -> suffix?

File: notebooks/truth_match_table.ipynb [hunk 5]
```diff
@@ -0,0 +1,436 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### What are \"best matches\"?\n",
+    "\n",
+    "Recall that `match_cat` is a \"left join\" of Object Table and Truth Table, so each object is assigned a \"best match\" truth soruce.\n",
+    "A \"best match\" truth soruce is the truth source that is within 1 arcsec of the object in consideration, and has the smallest magnitude difference.\n",
```
Comment (fjaviersanchez) [inline notebooks/truth_match_table.ipynb]: truth soruce -> truth source

File: notebooks/truth_match_table.ipynb [hunk 6]
```diff
@@ -0,0 +1,436 @@
...
+    "If there's no truth sources that are within 1 arcses of the object in consideration, or if the smallest magnitude difference is larger than 1 mag, \n",
+    "then the \"best match\" would be just the nearest neighbor. \n",
+    "\n",
+    "If a best match satifies the two creteria (separation < 1 arcsec and magnidute difference <  1 mag), we call it a \"good match\". Otherwise, it would just be the nearest neighbor. Note that a \"good match\" could still (in fact, is likely to) be the nearest neighbor. \n",
```
Comment (fjaviersanchez) [inline notebooks/truth_match_table.ipynb]: satifies -> satisfies

Comment (fjaviersanchez) [reply notebooks/truth_match_table.ipynb]: creteria -> criteria

Comment (fjaviersanchez) [reply notebooks/truth_match_table.ipynb]: magnidute -> magnitude

File: notebooks/truth_match_table.ipynb [hunk 7]
```diff
@@ -0,0 +1,436 @@
...
+   "outputs": [],
+   "source": [
+    "# Most good matches should also be nearest neighbors. \n",
+    "# And \"not good\" matches can be due to either large separation or large magnitude difference. \n",
+    "\n",
+    "# Let's define a few masks to distinguish these cases:\n",
+    "\n",
+    "dmag = np.abs(data[\"mag_r_truth\"] - data[\"mag_r_cModel\"])\n",
+    "\n",
+    "good_and_nn = data[\"is_good_match\"] & data[\"is_nearest_neighbor\"]\n",
```
Comment (fjaviersanchez) [inline notebooks/truth_match_table.ipynb]: Maybe add inline comment of what is `good_and_nn`?  Something like: ` # We select objects that are a `good` match and have been matched to their nearest neighbor`

File: notebooks/truth_match_table.ipynb [hunk 8]
```diff
@@ -0,0 +1,436 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In the figure above, the left panel shows the distribution of spatial separation for all best matches (note that y-axis is in log scale). \n",
+    "The vast majority of objects have matches within 1 arcsec, with a small tail goes to up to 5 arcsec. \n",
+    "\n",
+    "The right panel shows that, in each bin of spatial separation, the fraction of the four situations that we defined early. \n",
+    "The greens are \"good mathes\" (sep. < 1 arcsec and dmag < 1 mag), and we can see that the majority of \"good matches\" are also \n",
+    "nearest neighbor (darker green), especially if the separation is small. \n",
+    "The oranges are \"not good mathes\" (so that they are all nearest neighbors by definition).\n",
```
Comment (fjaviersanchez) [inline notebooks/truth_match_table.ipynb]: the oranges: the orange distributions.
good mathes -> good matches

File: notebooks/truth_match_table.ipynb [hunk 9]
```diff
@@ -0,0 +1,436 @@
...
+    "About only 20% of these nearest neighbors but not good mathes have close magnitude (dmag < 1 mag). \n",
```
Comment (fjaviersanchez) [inline notebooks/truth_match_table.ipynb]: mathes -> matches

**Review by fjaviersanchez — CHANGES_REQUESTED** [review]
Thanks @yymao. The notebook looks good to me and the second to last plot is very interesting! (the scatter plots of true - measured magnitudes for the different samples).

There are a few typos that need to be corrected but, otherwise, it looks great. Thank you again!

File: notebooks/truth_match_table.ipynb (hunk 3 above)
Comment (yymao) [reply notebooks/truth_match_table.ipynb]: I think "postfix" is more commonly used in software settings while "suffix" more common in daily use, so I added "(i.e., suffix)".

**Comment by yymao** [comment]
Thank you @fjaviersanchez -- I've fixed the typos and implemented your suggestion.

**Review by fjaviersanchez — APPROVED** [review]
Thank you for implementing the comments @yymao! Looks good to me.

**Comment by yymao** [comment]
Thank you @fjaviersanchez. I'll leave this PR open for a bit in case others have further comments. I also need to remove/update the custom paths when the download portol is set up.

**Pull request merged**